src/EventStore.Core/Services/SubscriptionsService.cs

```diff
@@ -267,7 +294,18 @@ public class SubscriptionsService : IHandle<SystemMessage.SystemStart>,
 					// resolve event if has not been previously resolved
 					resolvedEvent = pair = resolvedEvent ?? ResolveLinkToEvent(evnt, commitPosition);
 
-				subscr.Envelope.ReplyWith(new ClientMessage.StreamEventAppeared(subscr.CorrelationId, pair));
+				if (subscr.EventFilter.IsEventAllowed(evnt)) {
+					subscr.Envelope.ReplyWith(new ClientMessage.StreamEventAppeared(subscr.CorrelationId, pair));
+				}
+
+				subscr.CheckpointIntervalCurrent++;
+
+				if (subscr.CheckpointInterval != null &&
```

Comment:
Will the checkpoint still be reported on the client side even if the handler of the `StreamEventAppeared` throws?

Reply:
Yes, it will, after some exploration it seem that the event handler for `SubscribeToAllAsync` swallows exceptions and moves on. Given that, I'd expect the behaviour to be the same for the filtered method